Analysis/Tasks/PWGHF/HFJpsiCandidateSelector.cxx

```diff
@@ -0,0 +1,257 @@
+  template <typename T>
+  bool daughterSelection(const T& track)
+  {
+    if (track.charge() == 0) {
+      return false;
+    }
+    if (track.tpcNClsFound() == 0) {
+      return false; //is it clusters findable or found - need to check
+    }
+    return true;
+  }
+
+  /// Conjugate independent toplogical cuts
+  /// \param hfCandProng2 is candidate
+  /// \return true if candidate passes all cuts
+  template <typename T>
+  bool selectionTopol(const T& hfCandProng2)
+  {
+    auto candpT = hfCandProng2.pt();
+    int pTBin = getpTBin(candpT);
+    if (pTBin == -1) {
+      return false;
+    }
+
+    if (candpT < d_pTCandMin || candpT >= d_pTCandMax) {
+      return false; //check that the candidate pT is within the analysis range
+    }
+
+    return true;
+  }
+
+  /// Conjugate dependent toplogical cuts
+  /// \param hfCandProng2 is candidate
+  /// \param trackPositron is the track with the positron hypothesis
+  /// \return true if candidate passes all cuts for the given Conjugate
+  template <typename T1, typename T2>
+  bool selectionTopolConjugate(const T1& hfCandProng2, const T2& trackPositron, const T2& trackElectron)
+  {
+
+    auto candpT = hfCandProng2.pt();
+    int pTBin = getpTBin(candpT);
+    if (pTBin == -1) {
+      return false;
+    }
+
+    if (TMath::Abs(InvMassJpsi(hfCandProng2) - RecoDecay::getMassPDG(443)) > cuts[pTBin][0]) {
+      return false;
+    }
+
+    if ((trackElectron.pt() < cuts[pTBin][3]) || (trackPositron.pt() < cuts[pTBin][3])) {
+      return false; //cut on daughter pT
+    }
+    if (TMath::Abs(trackElectron.dcaPrim0()) > cuts[pTBin][1] || TMath::Abs(trackPositron.dcaPrim0()) > cuts[pTBin][1]) {
+      return false; //cut on daughter dca - need to add secondary vertex constraint here
+    }
+    if (TMath::Abs(trackElectron.dcaPrim1()) > cuts[pTBin][2] || TMath::Abs(trackPositron.dcaPrim1()) > cuts[pTBin][2]) {
+      return false; //cut on daughter dca - need to add secondary vertex constraint here
+    }
+
+    return true;
+  }
+
+  /// Check if track is ok for TPC PID
+  /// \param track is the track
+  /// \note function to be expanded
+  /// \return true if track is ok for TPC PID
+  template <typename T>
+  bool validTPCPID(const T& track)
+  {
+    if (TMath::Abs(track.pt()) < d_pidTPCMinpT || TMath::Abs(track.pt()) >= d_pidTPCMaxpT) {
+      return false;
+    }
+    //if (track.TPCNClsFindable() < d_TPCNClsFindablePIDCut) return false;
+    return true;
+  }
+
+  /// Check if track is compatible with given TPC Nsigma cut for a given flavour hypothesis
+  /// \param track is the track
+  /// \param nPDG is the flavour hypothesis PDG number
+  /// \param nSigmaCut is the nsigma threshold to test against
+  /// \note nPDG=11 electron
+  /// \return true if track satisfies TPC PID hypothesis for given Nsigma cut
+  template <typename T>
+  bool selectionPIDTPC(const T& track, int nPDG, int nSigmaCut)
+  {
+    double nSigma = 100.0; //arbitarily large value
```

Comment:
@nzardosh @DelloStritto. Should we make this consistent with the Lc task where we have changed to 1.0?

Reply:
The difference here is that the PID is being used here for the electrons but for the Lc it is not. The PID strategy is based on that of the D0 but with electrons instead of thee k,pi daughter. I think as a starting point this is probably ok? If not we can disable the PID for now by doing it like the Lc

Reply:
also the PID is only the TPC here. What do you think @ginnocen , do we keep the PID info for now or disable it until we properly look into it

Reply:
I would disable it at first.